1. Из колоды в 52 карты извлекаются случайным образом 4 карты. a) Найти вероятность того, что все карты – крести. б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

Всего карт масти крести в колоде 13 шт. 
для события А: 
1. Найти количество сочетаний для 4х карт крести  из 13 (x1)
2. Найти количество сочетаний для 4х карт из всей колоды (x2)
3. Поделить x1 на x2.

In [2]:
from math import factorial as fc

In [7]:
def combinations(n, k):
    return int(fc(n) / (fc(k)*fc(n-k)))

In [12]:
#1
x1 = combinations(13, 4);
x1

715

In [13]:
#2
x2 = combinations(52,4);
x2

270725

In [17]:
round((x1 / x2),4) 

0.0026

**Вероятность того, что все карты крести = 0.0026**

для события Б (не тузов в колоде 48):
возможны 4 результата при которых выпадет хотя бя один туз: 1 туз, 2 туза, 3 туза, 4 туза.

In [34]:
# Вариант с одним тузом
# Существует 4 варианта вытянуть один туз. 
x1 = combinations(48, 3);
x1

17296

In [35]:
# Комбинаций из 4х карт с одним тузом:
v1 = 4 * x1;
v1

69184

In [36]:
# Вариант с двумя тузами
x1 = combinations(4,2)
x2 = combinations(48,2)
v2 = x1*x2;
v2
# Комбинаций из 4х карт с двумя тузами:

6768

In [37]:
# Вариант с тремя тузами
x1 = combinations(4,3)
x2 = combinations(48,1)
v3 = x1*x2
v3
# Комбинаций их 4х карт с тремя тузами:

192

In [ ]:
# Вариант с четырьмя тузами:
1

In [41]:
# Просуммировать количество комбинаций для четырех вариантов и поделить на количество возможных комбинаций из 4х карт:
round((v1 + v2 + v3 + 1) / combinations(52,4), 3)

0.281

**Вероятность того, что среди 4х карт окажется хотя бы один туз = 0.281

2. На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

In [42]:
# Возможных комбинаций одновременного нажатия кнопок:
combinations(10,3)

120

In [44]:
#верная комбинация лишь одна. 
round((1 / 120), 3)

0.008

**Вероятность того, что человек, не знающий правильный код, откроет дверь с первого раза = 0.008**

3. В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?

In [51]:
round((combinations(9,3) / combinations(15,3)),3)

0.185

**Вероятность того, что все три детали окрашены = 0.185**

4. В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

In [56]:
round((1 / combinations(100,2)),5)

0.0002